setup-hooks.sh

```diff
@@ -0,0 +1,17 @@
+#!/bin/bash -eu
```

Comment:
Do you want to add some documentation to [druid/dev](https://github.com/apache/druid/tree/master/dev)?

We may want to give folks the option of where to install the checkstyle (i.e., pre-push or pre-commit).

Reply:
Added instruction to teamcity.md

Reply:
Actually I don't know where's a good place to add that

Reply:
Added to intellij-setup.md

Reply:
@maytasm3 did the intellij-setup.md change get pushed?

Reply:
@jon-wei  Lost my branch. Opened a new PR with the intellij-setup.md change.